### DISCUSSION TITLE
Conditional Content more fine-grained

### DISCUSSION THREAD

**Discussion by JanPalasek** [quarto-dev/quarto-cli#7466] [Q&A] [answered]
### Description

I would like to hide Conditional Content for html output, but show it in revealjs presentations. However, according to docs, I can't do that now: https://quarto.org/docs/authoring/conditional.html.

Typical usecase: I have a presentation in revealjs. For students, I output html document so they can later check out what was in the presentation. This output document needs to be e.g. without lecturer notes, and possibly without other presentation-specific things. However since format matching "html" (in `when="html"`) matches both html and revealjs, I can't do that now.

To circle around this problem, I use my custom-defined hiding classes that I only apply for the output html document. However, since Quarto already has a conditional content, it would be nice to have it integrated inside. For example there could be new `revealjs` format, and I could specify `when="html" unless="revealjs"`.

**Comment by mcanouil** [comment, marked as answer]
> However, according to docs, I can't do that now

Where is this stated exactly as it is not true?

See:
- https://github.com/quarto-dev/quarto-cli/discussions/3215#discussioncomment-4095809

**Reply by JanPalasek** [reply]
@mcanouil Thanks for the answer. This solves the issue.

After reading it again, it isn't directly stated. However I believe it is heavily implied. The entire documentation page uses html and pdf. Then it states that those are aliases. Since "html" and "pdf" (the only formats used in the page) are aliases, I assumed that only those formats (resp. aliases) are supported.

Maybe it would be less confusing to call every format that we specify (in the when-format and unless-format) an alias and list them all? Or use format (that is not alias) in the documentation?

**Reply by mcanouil** [reply]
> Maybe it would be less confusing to call every format that we specify (in the when-format and unless-format) an alias and list them all?

This is not possible, as there are no exhaustive list of formats.
For instance, if you interpret the documentation as "since there is no example for revealjs then it does not work" it would mean that this does not work for any custom format extension which is far from true, but it's not possible to have examples of those custom formats.
Examples are examples.
The documentation shows a table for format matching pattern.

**Reply by JanPalasek** [reply]
@mcanouil Isn't worth it to imply that there are other options than the listed ones? It is really not obvious that `html` is alias (which by the name should be a format), while there also exists e.g. `revealjs` as a format.

**Reply by mcanouil** [reply]
That's the column names, isn't it explicit enough?
![image](https://github.com/quarto-dev/quarto-cli/assets/8896044/0df698b4-9ef1-41d7-8261-b293bfc8ca29)

**Reply by JanPalasek** [reply]
In my opinion, no. The fact that you have alias `html` and `pdf` automatically leads me to thinking that you can specify only aliases in `when-format`. But that's only my pov.

**Reply by mcanouil** [reply]
I see, but as far as I am aware, you are the first to read this like that.
The options are called `when-format` and `unless-format`, if those were only accepting aliases the names would be very bad.
Also the documentation refers to "format" when explaining the concept of the options while there is a sentence saying that there is more than regular formats, you have aliases to match the output format:
![image](https://github.com/quarto-dev/quarto-cli/assets/8896044/4ebab8e9-5b62-4662-b76b-5c84f73a1282)

Do you have a concrete proposal here?

**Reply by JanPalasek** [reply]
@mcanouil In the documentation page, I suggest to use `revealjs` instead of `pdf`. For example:

```
::: {.content-visible unless-format="revealjs"}

Will not appear in revealjs.

:::

```

Or maybe use docx instead of revealjs.

I think it would be useful to have an example with something with a format (not alias).

**Reply by mcanouil** [reply]
Well `pdf` is a format and an alias.

**Reply by JanPalasek** [reply]
One example with something that isn't an alias.

**Reply by mcanouil** [reply]
I'll make a PR with the revealjs/html case from my previous answer.
This will also show a when/unless example.

**Reply by mcanouil** [reply]
FYI:
- quarto-dev/quarto-web#910

**Reply by JanPalasek** [reply]
Looks great. Thx.

**Comment by harrelfe** [comment]
A far simpler approach is dual-use notes using standard `Quarto`.  Multiple instructors have found this to be well received by students.  Put side notes a `.aside` notes and use the long-form notes without somewhat distracting slide pagination.  Merely by hitting `curl +` a few times the long form notes are converted into quite presentable slides.  And long-term maintenance of the material is easy.  

Besides the use of marginal notes, standard `Quarto` collapsed blocks are very effective for housing comments and further details.

**Reply by mcanouil** [reply]
Note that this is for a particular use case where there is a need to get content from html/revealjs to the other formats.
This won't work if the use case is to have format specific content.
